### PR TITLE
Center map on location selection

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

When an origin or destination is set, the map will now center to that point. If both are set, the map will show both. New refactored code makes this happen more efficiently with fewer frame drops.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/map.js [hunk 1]
```diff
@@ -22,9 +22,7 @@ export function setMapCenter(map /* MapRef */, location, zoom) {
     const { lat, lon } = location
     if (map && !isNaN(lat) && !isNaN(lon)) {
       const center = [lon, lat]
-      // Note: passing an undefined zoom to map.jumpTo will trigger a "can't invert matrix" error.
-      const jumpParams = zoom !== undefined ? { center, zoom } : { center }
-      map.jumpTo(jumpParams)
```
Comment (binh-dam-ibigroup) [inline lib/actions/map.js:27]: `zoom` becomes unused and can be removed from the function parameters.

**Review by binh-dam-ibigroup — APPROVED** [review]
Nice

File: lib/components/app/responsive-webapp.js [hunk 2]
```diff
@@ -97,9 +96,19 @@ class ResponsiveWebapp extends Component {
...
       if (isMobile() && query.from === null) {
         setLocationToCurrent({ locationType: 'from' }, intl)
-        setMapCenter(map, pt, initZoomOnLocate)
+        setMapCenter(map, pt)
       }
+    } else if (query.from && query.to) {
+      map?.fitBounds([query.from, query.to], {
+        duration: 500,
+        padding: getFitBoundsPadding(map, 0.2)
+      })
```
Comment (amy-corson-ibigroup) [inline lib/components/app/responsive-webapp.js:105]: I think something we discussed was if both location fields are filled, set the nearby center to the destination location? Setting the nearby center to the center of the trip is what we were looking to change

**Review by amy-corson-ibigroup — CHANGES_REQUESTED** [review]
Nice! Just a question about the behavior when both location fields are set.

File: lib/components/app/responsive-webapp.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/app/responsive-webapp.js:105]: That happens in a different PR. This PR only handles map panning

**Pull request merged**